### PR TITLE
Modify rule S7027: Update description and align with S7091

### PR DESCRIPTION
**Pull request by kaufco**

### CODE REVIEW

File: rules/S7027/java/rule.adoc [hunk 1]
```diff
@@ -1,18 +1,19 @@
-== Why is this an issue?
-
-Circular dependencies happen when two classes, directly or indirectly, reference each other in their definitions.
+This rule reports circular dependencies between classes that all reside in the same package.
```
Comment (irina-batinic-sonarsource) [inline rules/S7027/java/rule.adoc:1]: This rule identifies circular dependencies between classes within the same package.
```suggestion
This rule identifies circular dependencies between classes within the same package.
```

File: rules/S7027/java/rule.adoc [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
 
-This cyclical relationship means class boundaries cannot be expressed as an intuitive hierarchy. This degrades your code’s structure and makes it harder to manage and extend over time.
+== Why is this an issue?
 
-These dependencies can lead to unexpected behavior, which complicates debugging, and makes your code more error-prone.
+Circular dependencies occur when two classes reference each other directly or indirectly.
```
Comment (irina-batinic-sonarsource) [inline rules/S7027/java/rule.adoc:5]:
```suggestion
Circular dependencies occur when two classes reference each other, either directly or indirectly.
```

File: rules/S7027/java/rule.adoc [hunk 3]
```diff
@@ -1,18 +1,19 @@
...
+This means that the classes' dependency structure does not represent an intuitive hierarchy, which makes it harder to understand and maintain.
 
 === What is the potential impact?
 
-Circular dependencies complicate code maintenance, making it difficult to change one class without inadvertently affecting others and increasing the risk of defects.
+Circular dependencies increase the architectural complexity of the code, reducing its readability, extensibility, and maintainability.
+As a project grows, circular dependencies tend to be a seed for additional circular dependencies and further increase in architectural complexity that adds to the technical depth.
```
Comment (irina-batinic-sonarsource) [inline rules/S7027/java/rule.adoc:11]:
```suggestion
As the project grows, circular dependencies often lead to more circular dependencies, further complicating the architecture and increasing technical debt.
```

File: rules/S7027/java/rule.adoc [hunk 4]
```diff
@@ -1,18 +1,19 @@
...
+Over time, untangling these dependencies becomes increasingly difficult.
 
-In some cases, they can even prevent your code from compiling or cause runtime failures. As your project grows, these cycles become harder to untangle, further degrading the codebase and increasing technical debt.
+=== Exceptions
 
-This structural issue can significantly hinder your ability to adapt and improve the code in the future.
+Circular dependencies between an inner class and its enclosing class, or between inner classes enclosed by the same class, are not reported.
```
Comment (irina-batinic-sonarsource) [inline rules/S7027/java/rule.adoc:16]:
```suggestion
Circular dependencies between an inner class and its outer class, or between inner classes within the same enclosing class, are not flagged as issues.
```

File: rules/S7027/java/metadata.json [hunk 5]
```diff
@@ -1,5 +1,5 @@
 {
-  "title": "Cyclic dependencies between classes should be resolved",
+  "title": "Circular dependencies between classes in the same package should be resolved",
```
[the remark this answers is not in the document]
Comment (kaufco) [reply rules/S7027/java/metadata.json:2]: Fixed.

**Review by irina-batinic-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**